Falco Rules/Conditions update

raviranjanelastisys opened the pull request:
**What type of PR is this?**

/kind rule-update

**Any specific area of the project related to this PR?**

/area rules

**What this PR does / why we need it:**

Improving security posture and intrusion detection &  it needs careful tuning to ensure good signal-to-noise ratio. Falco alerts should eventually become alert-worthy. This PR consists in filtering a few "obvious" false-positives.
1. Based on our investigations and finding , Contact K8S API Server From Container" should be ignored for container images which are known "controllers".
2. Based on our investigations and finding , updated the rules to `write_etc_common` , `Launch Package Management Process in Container` , `falco_privileged_images`, `k8s_containers` , `postgres_running_wal_e` & `falco_sensitive_mount_containers`.

**Does this PR introduce a user-facing change?**

`NONE`

poiana commented:
@raviranjanelastisys: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/falco/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [26daf95](https://github.com/falcosecurity/falco/commits/26daf9500d34ed1a18a4d34266eee9837f2da0d5) Rules update

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/falco/pull/1981#" title="Author self-approved">raviranjanelastisys</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **kaizhe** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @kaizhe` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Ffalco).

<details open>
Needs approval from an approver in each of these files:

- **[rules/OWNERS](https://github.com/falcosecurity/falco/blob/master/rules/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["kaizhe"]} -->

poiana commented:
Welcome @raviranjanelastisys! It looks like this is your first PR to falcosecurity/falco 🎉

raviranjanelastisys commented:
/release-note-none

raviranjanelastisys commented:
/assign @kaizhe

leogr commented on rules/falco_rules.yaml:
```diff
@@ -2428,6 +2437,12 @@
     and package_mgmt_procs
     and not package_mgmt_ancestor_procs
     and not user_known_package_manager_in_container
+  exceptions:
+   - name: kube-proxy
+     fields: [proc.cmdline, container.image.repository]
+     comps: [startswith, =]
+     values:
+       - [update-alternat, k8s.gcr.io/kube-proxy]
```
We are not using _exceptions_ in the default ruleset (after a long discussion in the community that happened 1 year ago). I'd suggest rewriting that exclusion by using macros.

raviranjanelastisys replied:
@leogr This has been fixed now

leogr commented:
Hey @raviranjanelastisys 

I believe there's also some syntax error, see the failing test :point_down: 
https://app.circleci.com/pipelines/github/falcosecurity/falco/2453/workflows/d3326c2e-2279-44cc-8bf2-2b98e55f933a/jobs/22611

Could you take a look?

Thanks :pray:

raviranjanelastisys commented:
> Hey @raviranjanelastisys
> 
> I believe there's also some syntax error, see the failing test point_down https://app.circleci.com/pipelines/github/falcosecurity/falco/2453/workflows/d3326c2e-2279-44cc-8bf2-2b98e55f933a/jobs/22611
> 
> Could you take a look?
> 
> Thanks pray

@leogr Thanks !! But i don't see any syntax error !!

leogr commented on rules/falco_rules.yaml:
```diff
@@ -1253,6 +1253,7 @@
     and not mcafee_writing_cma_d
     and not avinetworks_supervisor_writing_ssh
     and not multipath_writing_conf
+    and not calico-node
```
```suggestion
    and not calico_node
```
This should be the correct macro name

raviranjanelastisys replied:
@leogr Thanks !! This has been updated now !!

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/falco/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [6979c9d](https://github.com/falcosecurity/falco/commits/6979c9da0ea7d9c939d4901b5e88dbbc3572bcef) fix

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/falco/pull/1981#" title="Author self-approved">raviranjanelastisys</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **kaizhe** after the PR has been reviewed.

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Ffalco).

<details open>
Needs approval from an approver in each of these files:

- **[rules/OWNERS](https://github.com/falcosecurity/falco/blob/master/rules/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["kaizhe"]} -->

raviranjanelastisys commented:
> 

Thanks @leogr This has been updated now

raviranjanelastisys commented:
Any update ?

leogr commented:
/milestone 0.33.0

raviranjanelastisys commented:
Any update

jasondellaluce commented on rules/falco_rules.yaml:
```diff
@@ -2428,6 +2437,12 @@
     and package_mgmt_procs
     and not package_mgmt_ancestor_procs
     and not user_known_package_manager_in_container
+  exceptions:
```
It looks like these changes still contain exceptions, which as @leogr mentioned are not supported by the community in the default rulesets. Moreover, your branch probably needs to be rebased due to some recent changes on master.

raviranjanelastisys replied:
@jasondellaluce Thanks , This has been fixed now

Kaizhe commented:
@darryk10 please take a look!

darryk10 commented on rules/falco_rules.yaml:
```diff
@@ -1471,7 +1472,8 @@
 - macro: postgres_running_wal_e
-  condition: (proc.pname=postgres and proc.cmdline startswith "sh -c envdir /etc/wal-e.d/env /usr/local/bin/wal-e")
+  condition: ((proc.pname=postgres and proc.cmdline startswith "sh -c envdir /etc/wal-e.d/env /usr/local/bin/wal-e") or
+              (proc.pname=postgres and proc.cmdline startswith "sh -c envdir \"/run/etc/wal-e.d/env\" wal-g wal-push"))
```
```suggestion
              (proc.pname=postgres and proc.cmdline startswith "sh -c envdir \"/run/etc/wal-e.d/env\" wal-g wal-push"))
```
```suggestion
              (proc.pname=postgres and proc.cmdline startswith "sh -c envdir \"/run/etc/wal-e.d/env\" wal-g wal-push"))
```
```suggestion
              (proc.pname=postgres and (proc.cmdline startswith "sh -c envdir /etc/wal-e.d/env /usr/local/bin/wal-e" or proc.cmdline startswith "sh -c envdir \"/run/etc/wal-e.d/env\" wal-g wal-push"))
```

raviranjanelastisys replied:
@darryk10 Thanks this has been fixed now

darryk10 commented on rules/falco_rules.yaml:
```diff
@@ -2365,7 +2369,12 @@
-     public.ecr.aws/falcosecurity/falco)
+     public.ecr.aws/falcosecurity/falco, velero/velero,
+     quay.io/jetstack/cert-manager-cainjector, weaveworks/kured,
+     ingress-controller/controller, quay.io/prometheus-operator/prometheus-operator,
```
I haven't found `ingress-controller/controller` as a public image. Can you check this or remove it pls.

darryk10 commented on rules/falco_rules.yaml:
```diff
@@ -1881,7 +1884,8 @@
               container.image.repository in (falco_sensitive_mount_images) or
-              container.image.repository startswith quay.io/sysdig/)
+              container.image.repository startswith quay.io/sysdig/ or
+              container.image.repository startswith k8scloudprovider/cinder-csi-plugin)
```
```suggestion
              container.image.repository=k8scloudprovider/cinder-csi-plugin)
```

raviranjanelastisys replied:
@darryk10 Thanks this has been fixed now

darryk10 reviewed (commented):
I proposed an optimisation for the condition to make it a bit more readable and other minor changes.

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/falco/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [3984987](https://github.com/falcosecurity/falco/commits/39849873fd54c100b3a6ad9482fa8ce6d6048e7b) fix

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

darryk10 reviewed (commented):
Thanks for the cooperation on that. I just added a couple of more suggestions on the macro name but overall it looks good to me.

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/falco/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [53c48d6](https://github.com/falcosecurity/falco/commits/53c48d67d51ef4fb08e0fad89eb2928c238c4bbd) fix

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

raviranjanelastisys commented on rules/falco_rules.yaml:
```diff
@@ -2419,6 +2427,11 @@
   condition: (never_true)
 
 # Container is supposed to be immutable. Package management should be done in building the image.
+- macro: pkg_mgmt_in_kube-proxy
```
@darryk10 Thanks this has been fixed now

darryk10 reviewed (approved):
Thanks for the PR and all changes

jasondellaluce commented:
@raviranjanelastisys some of the tests are failing for this PR. I think rebasing this on master could be enough to fix it, because some changes have been applied to our CI over the past few weeks. Would you mind rebasing this on master?

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/falco/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [a652118](https://github.com/falcosecurity/falco/commits/a652118f381f718ca6117ebc7940a07bdc5b8a17) Merge branch 'update-falco-rule' of github.com:elastisys/falco-upstream into update-falco-rule

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/falco/pull/1981#" title="Author self-approved">raviranjanelastisys</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **kris-nova** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @kris-nova` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Ffalco).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/falco/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["kris-nova"]} -->

poiana commented:
@raviranjanelastisys: Adding label `do-not-merge/contains-merge-commits` because PR contains merge commits, which are not allowed in this repository.
Use `git rebase` to reapply your commits on top of the target branch. Detailed instructions for doing so can be found [here](https://git.k8s.io/community/contributors/guide/github-workflow.md#4-keep-your-branch-in-sync).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

jasondellaluce commented:
@raviranjanelastisys merge commits are not allowed by @poiana and our CI in order maintain a linear git history. The way you should do it is through a rebase, not a branch merging.

raviranjanelastisys commented:
> > > e commits are not all
> > 
> > 
> > yeah agreed !! I think i did mistake while rebasing and more commit got added . Now trying to revert back to my old commits !! and then rebase to master
> 
> @jasondellaluce is there a way i can revert back to my old commits !! I have only updated the falco rules files and nothing else

These are not my changes 
![image](https://user-images.githubusercontent.com/75951259/176644193-9c1902b1-b433-4eab-b49b-063a4b17f60c.png)

jasondellaluce commented:
You need to rebase your branch on `master`. The additional commits you see are the result of merging with the `master` branch instead of rebasing agains it. A solution would be to rebase and drop all the additional commits and the merge commit. Take a look here: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

If you can't find a way, and give me explicit consensus here in a comment, I can rebase and push on your branch for you.

raviranjanelastisys commented:
> You need to rebase your branch on `master`. The additional commits you see are the result of merging with the `master` branch instead of rebasing agains it. A solution would be to rebase and drop all the additional commits and the merge commit. Take a look here: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History
> 
> If you can't find a way, and give me explicit consensus here in a comment, I can rebase and push on your branch for you.

@jasondellaluce Thank you so much !! Yes i give my consent and request you to rebase and push on my branch . I don't want to mess it up more on commits !! Thank you much very much

jasondellaluce commented:
Apparently, my privileges as maintainer are not enough to push to your fork 😢 
However, here's an easy way you can do this. I will assume you've set the `falcosecurity/falco` repository as `upstream` in your local repo.

First, you have to drop all the extra commits created during the merge. There's 9 commits you need to get rid of. Switch to your branch with `git checkout update-falco-rule`, then run `git rebase -i HEAD~10`. You'll a list containing the last 10 commits of your branch in your terminal's text editor. Each commit will have a `pick` option on the left. Change the `pick` into `d` (which stands for _drop_) for the last 9 commits (leave `pick` your first commit with hash `e651044`). Then save the text file and exit from the editor. You should see an output like `Successfully rebased and updated refs/heads/update-falco-rule`.

Second, you need to rebase against master from `falcosecurity/falco`. `git rebase upstream/master` would do the job here. Again, you should see something like `Successfully rebased and updated refs/heads/update-falco-rule`.

Finally, you need to push-force to your branch: `git push --force-with-lease update-falco-rule`.

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/falco/pull/1981#" title="Author self-approved">raviranjanelastisys</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mstemm** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mstemm` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Ffalco).

<details open>
Needs approval from an approver in each of these files:

- **[rules/OWNERS](https://github.com/falcosecurity/falco/blob/master/rules/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["mstemm"]} -->

raviranjanelastisys commented:
> Apparently, my privileges as maintainer are not enough to push to your fork cry However, here's an easy way you can do this. I will assume you've set the `falcosecurity/falco` repository as `upstream` in your local repo.
> 
> First, you have to drop all the extra commits created during the merge. There's 9 commits you need to get rid of. Switch to your branch with `git checkout update-falco-rule`, then run `git rebase -i HEAD~10`. You'll a list containing the last 10 commits of your branch in your terminal's text editor. Each commit will have a `pick` option on the left. Change the `pick` into `d` (which stands for _drop_) for the last 9 commits (leave `pick` your first commit with hash `e651044`). Then save the text file and exit from the editor. You should see an output like `Successfully rebased and updated refs/heads/update-falco-rule`.
> 
> Second, you need to rebase against master from `falcosecurity/falco`. `git rebase upstream/master` would do the job here. Again, you should see something like `Successfully rebased and updated refs/heads/update-falco-rule`.
> 
> Finally, you need to push-force to your branch: `git push --force-with-lease update-falco-rule`.
> 
> An alternative would also be to copy the diff from your original commit and open a new PR from a new branch based on master (and close this PR).

@jasondellaluce Thank you very much for detailed explanations !! Really appreciated :) !! Will use this as a reference for future PR too :) !! 
@darryk10 Request you to approve again :)

darryk10 reviewed (dismissed):
LGTM

poiana commented:
@darryk10: changing LGTM is restricted to collaborators

<details>

In response to [this](https://github.com/falcosecurity/falco/pull/1981#pullrequestreview-1027146910):

>LGTM


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

jasondellaluce commented:
Looks good now @raviranjanelastisys! I think the integration tests are failing because the `calico_node` macro can't be resolved. The reason is that `calico_node` is defined at line _after_ the `write_etc_common` macro that uses it. Macros have visibility only on macros defined before them. You can fix this easily with another commit by just moving the `calico_node` macro definition up before `write_etc_common`.

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/falco/blob/master/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [31ce12d](https://github.com/falcosecurity/falco/commits/31ce12d001c7bf0a9cfc500bd06fbe3b04e5ad47) fix

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

raviranjanelastisys commented:
@jasondellaluce @darryk10 Thank you so much !! All the tests are passed now !! Please approve again and so i can merge it :)

poiana commented:
LGTM label has been added.  <details>Git tree hash: 5d696e784400d277886abd45629259b74e846301</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/falco/pull/1981#pullrequestreview-1027305429" title="Approved">jasondellaluce</a>*, *<a href="https://github.com/falcosecurity/falco/pull/1981#" title="Author self-approved">raviranjanelastisys</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Ffalco).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[rules/OWNERS](https://github.com/falcosecurity/falco/blob/master/rules/OWNERS)~~ [jasondellaluce]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

raviranjanelastisys commented:
@darryk10 Your +1 is required !! Thanks

darryk10 reviewed (approved):
LGTM! 
Thank again @raviranjanelastisys!

poiana commented:
@darryk10: changing LGTM is restricted to collaborators

<details>

In response to [this](https://github.com/falcosecurity/falco/pull/1981#pullrequestreview-1027493268):

>LGTM! 
>Thank again @raviranjanelastisys!


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

jasondellaluce commented:
/unhold

FedeDP reviewed (approved):
/approve
